instrumentation/undertow/javaagent/src/test/groovy/UndertowServerTest.groovy

```diff
@@ -66,4 +93,65 @@ class UndertowServerTest extends HttpServerTest<Undertow> implements AgentTestTr
   String expectedServerSpanName(ServerEndpoint endpoint) {
     return "HTTP GET"
   }
+
+  @Unroll
+  def "test send response"() {
+    setup:
+    def url = HttpUrl.get(address.resolve("sendResponse")).newBuilder().build()
+    def request = request(url, "GET", null).build()
+    Response response = client.newCall(request).execute()
+
+    expect:
+    response.code() == 200
+    response.body().string().trim() == "sendResponse"
+
+    and:
+    assertTraces(1) {
+      trace(0, 2) {
+        it.span(0) {
+          hasNoParent()
+          name "HTTP GET"
+
+          event(0) {
+            eventName "before-event"
+          }
+          event(1) {
+            eventName "after-event"
+          }
+        }
+        basicSpan(it, 1, "sendResponse", span(0))
+      }
+    }
+  }
+
+  @Unroll
+  def "test send response with exception"() {
+    setup:
+    def url = HttpUrl.get(address.resolve("sendResponseWithException")).newBuilder().build()
+    def request = request(url, "GET", null).build()
+    Response response = client.newCall(request).execute()
+
+    expect:
+    response.code() == 200
+    response.body().string().trim() == "sendResponseWithException"
+
+    and:
+    assertTraces(1) {
+      trace(0, 2) {
+        it.span(0) {
+          hasNoParent()
+          name "HTTP GET"
+
+          event(0) {
+            eventName "before-event"
+          }
+          event(1) {
+            eventName "after-event"
+          }
+          errorEvent(Exception, "exception after sending response", 2)
```

Comment:
can you add `attributes` validation also? e.g. would be good to see that we're capturing `200` even though also capturing the exception